crates/conformance-test-utils/src/hive/genesis.rs

```diff
@@ -23,6 +36,125 @@ impl HiveGenesisConfig {
     }
 }
 
+/// Convert Hive Genesis Config to Madara Genesis Config
+///
+/// This function will:
+/// 1. Load the Madara genesis file
+/// 2. Construct a Starknet test sequencer
+/// 3. Declare Kakarot contracts
+/// 4. Add Kakarot contracts to Loader
+/// 5. Add Hive accounts to Loader
+/// 6. Serialize Loader to Madara genesis file
+pub async fn serialize_hive_to_madara_genesis_config(hive_genesis: HiveGenesisConfig) -> Result<(), IoError> {
+    // Load the Madara genesis file
+    let mut loader: GenesisLoader =
+        serde_json::from_str(&read_file_to_string("crates/conformance-test-utils/src/madara/genesis.json"))
+            .expect("Failed to load Madara genesis file");
+
+    // Get compiled path of contracts on loader
+    let mut compiled_path = PathBuf::from("");
+    if let ContractClass::Path { path, .. } = &loader.contract_classes[0].1 {
+        compiled_path = PathBuf::from(path);
+
+        // Remove filename
+        compiled_path.pop();
+    }
+
+    // Declare Kakarot contracts
+    let class_hashes = compute_kakarot_contracts_class_hash();
+
+    // { contract : (address, class_hash) }
+    let mut kakarot_contracts = HashMap::<String, (FieldElement, FieldElement)>::new();
+    // Add Kakarot contracts Contract Classes to loader
+    // Vec so no need to sort
+    class_hashes.iter().enumerate().for_each(|(index, (filename, class_hash))| {
+        loader.contract_classes.push((
+            HexFelt(*class_hash),
+            ContractClass::Path {
+                // Add the compiled path to the Kakarot contract filename
+                path: compiled_path.join(filename).with_extension("json").into_os_string().into_string().unwrap(), /* safe unwrap,
+                                                                                             * valid path */
+                version: 0,
+            },
+        ));
+        // Add Kakarot contracts (address, class_hash) to Kakarot Contracts HashMap
+        // Remove .json from filename to get contract name
+        // 36865 is over 0x9000
+        kakarot_contracts.insert(filename.replace(".json", ""), (FieldElement::from(36865 + index), *class_hash));
+    });
+
+    // Get Kakarot contracts address and proxy class hash
+    let kakarot_address = kakarot_contracts.get("kakarot").unwrap().0;
+    let account_proxy_class_hash = kakarot_contracts.get("proxy").unwrap().1;
+
+    // Add Kakarot contracts to Loader
+    // Convert the HashMap to Vec and sort by key to ensure deterministic order
```

Comment:
So you are using the name of the contract to store it deterministically? Would it change anything to sort by address?

Reply:
Good catch, I think it would make sense to sort by address as we are loading it in `contracts` 👍